controller/src/main.rs

```diff
@@ -1,24 +1,28 @@
 use controller_lib::external_api;
 use controller_lib::internal_api;
 
+mod config;
+
 #[tokio::main]
 async fn main() -> std::io::Result<()> {
-    let grpc_server_addr: String = "0.0.0.0:50051".to_string();
-    let grpc_server_num_workers: usize = 1;
-
-    let http_server_addr: String = "0.0.0.0:3000".to_string();
-    let http_server_num_workers: usize = 1;
-
     // Init Logger
     env_logger::init();
 
+    let config: config::KudoControllerConfig = confy::load_path("controller.conf").unwrap();
```

Comment:
You should return a result here.

Reply:
But this one result in the same consequence as it is in the main function, right ? (I changed it anyway directly by doing a force push on commit d8bf76e)

Reply:
Consequences are different. `unwrap` panics, while a main returning an error is handled more gracefully (proper stack dump, etc).